assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -12955,92 +12955,124 @@ private void initializeAssignment(SessionState state)
+		int dueDateOffset = ServerConfigurationService.getInt("assignment.duedate", 604800);
+		int acceptUntilDateOffset = ServerConfigurationService.getInt("assignment.acceptuntildate", 691200);
+		int peerEvaluationDateOffset = ServerConfigurationService.getInt("assignment.peerevaluationdate", 1296000);
+
 		// get current time
 		Time t = TimeService.newTime();
-		TimeBreakdown tB = t.breakdownLocal();
-		int month = tB.getMonth();
-		int day = tB.getDay();
-		int year = tB.getYear();
-		int hour = tB.getHour();
-		int minute = tB.getMin();
-
-		// set the visible time to be 12:00 PM
-                if (Boolean.valueOf(ServerConfigurationService.getBoolean("assignment.visible.date.enabled", false))) {                
-                    state.setAttribute(NEW_ASSIGNMENT_VISIBLEMONTH, Integer.valueOf(month));
-                    state.setAttribute(NEW_ASSIGNMENT_VISIBLEDAY, Integer.valueOf(day));
-                    state.setAttribute(NEW_ASSIGNMENT_VISIBLEYEAR, Integer.valueOf(year));
-                    state.setAttribute(NEW_ASSIGNMENT_VISIBLEHOUR, Integer.valueOf(12));
-                    state.setAttribute(NEW_ASSIGNMENT_VISIBLEMIN, Integer.valueOf(0));
-                    state.setAttribute(NEW_ASSIGNMENT_VISIBLETOGGLE, false);
-                }
-                
-		// set the open time to current time
+		TimeBreakdown tB;
+		int month;
+		int day;
+		int year;
+		int hour;
+		int minute;
+		
+		if (Boolean.valueOf(ServerConfigurationService.getBoolean("assignment.visible.date.enabled", false))) {
+			t.setTime(t.getTime() + visibleDateOffset * 1000);
+			tB = t.breakdownLocal();
+			month = tB.getMonth();
+			day = tB.getDay();
+			year = tB.getYear();
+			hour = tB.getHour();
+			minute = tB.getMin();
+			state.setAttribute(NEW_ASSIGNMENT_VISIBLEMONTH, Integer.valueOf(month));
+			state.setAttribute(NEW_ASSIGNMENT_VISIBLEDAY, Integer.valueOf(day));
+			state.setAttribute(NEW_ASSIGNMENT_VISIBLEYEAR, Integer.valueOf(year));
+			state.setAttribute(NEW_ASSIGNMENT_VISIBLEHOUR, Integer.valueOf(hour));
+			state.setAttribute(NEW_ASSIGNMENT_VISIBLEMIN, Integer.valueOf(minute));
+			state.setAttribute(NEW_ASSIGNMENT_VISIBLETOGGLE, false);
+		}
+
+		// open date is shifted forward by the offset
+		t.setTime(t.getTime() + (openDateOffset - visibleDateOffset) * 1000);
+		tB = t.breakdownLocal();
+		month = tB.getMonth();
+		day = tB.getDay();
+		year = tB.getYear();
+		hour = tB.getHour();
+		minute = tB.getMin();
+
 		state.setAttribute(NEW_ASSIGNMENT_OPENMONTH, Integer.valueOf(month));
 		state.setAttribute(NEW_ASSIGNMENT_OPENDAY, Integer.valueOf(day));
 		state.setAttribute(NEW_ASSIGNMENT_OPENYEAR, Integer.valueOf(year));
-		state.setAttribute(NEW_ASSIGNMENT_OPENHOUR, Integer.valueOf(12));
-		state.setAttribute(NEW_ASSIGNMENT_OPENMIN, Integer.valueOf(0));
-		state.setAttribute(NEW_ASSIGNMENT_OPENHOUR, hour);
-		state.setAttribute(NEW_ASSIGNMENT_OPENMIN, minute);
+		state.setAttribute(NEW_ASSIGNMENT_OPENHOUR, Integer.valueOf(hour));
```

Comment:
I would have changes this so that we don't have the `month/day/year` variables because:
- they are only read once for each input.
- the variable isn't clear what it's currently set to
- the variables get updated in different parts of the method to mean different things.

I'd have had a specific `TimeBreakdown` for each input and done something like:
````
TimeBreadkdown open = TimeService.newTime(now.getTime +(openDateOffset - visibleDateOffset) * 1000 ).breakdownLocal();

...[snipped]..

state.setAttribute(NEW_ASSIGNMENT_OPENHOUR, Integer.valueOf(open.getHour());
````